receiver/libhoneyreceiver/internal/libhoneyevent/libhoneyevent.go

```diff
@@ -88,7 +94,25 @@ func (l *LibhoneyEvent) DebugString() string {
 
 // SignalType returns the type of signal this event represents. Only log is implemented for now.
 func (l *LibhoneyEvent) SignalType() (string, error) {
-	return "log", nil
+	if sig, ok := l.Data["meta.signal_type"]; ok {
+		switch sig {
+		case "trace":
+			if atype, ok := l.Data["meta.annotation_type"]; ok {
+				if atype == "span_event" {
+					return "span_event", nil
+				} else if atype == "link" {
+					return "span_link", nil
+				}
+				return "span", errors.New("invalid annotation type, but probably a span")
+			}
+			return "span", nil
+		case "log":
+			return "log", nil
+		default:
+			return "log", errors.New("invalid meta.signal_type")
+		}
+	}
+	return "log", errors.New("missing meta.signal_type and meta.annotation_type")
```

Comment:
```suggestion
	return "", errors.New("missing meta.signal_type and meta.annotation_type")
```

Reply:
It _should_ be able to parse anything that comes in as a log even if it hasn't been explicitly set as a log.  Adding the error and allowing the parser to still proceed seems like the right balance here. Otherwise, I'd rather remove the error and leave it as a default of "log" whenever it fails to find trace-like clues.

Reply:
Returning an error and using a valid value isn't typical, so if we are purposefully falling back to log whenever possible to guarantee ingestion I agree with removing the returned error.  Instead we can write a warning log inside this function.